gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposePayees.ts

```diff
@@ -80,35 +94,60 @@ export default class ProposePayees extends SolanaCommand {
       'Every payee needs to have a valid token recipient address',
     )
 
-    const proposalInfo = await program.account.proposal.fetch(proposal)
-    const payeeByTransmitter = input.operators.reduce(
-      (agg, operator) => ({
-        ...agg,
-        [new PublicKey(operator.transmitter).toString()]: new PublicKey(operator.payee),
-      }),
-      {},
-    )
-
-    // Set the payees in the same order the oracles are saved in the proposal. The length of the payees need to be same as the oracles saved
+    // Set the payees in the same order the oracles are saved in the proposal
+    // The length of the payees need to be same as the oracles saved
+    const proposal = new PublicKey(input.proposalId)
+    const proposalInfo = await this.program.account.proposal.fetch(proposal)
     const payees = proposalInfo.oracles.xs
       .slice(0, proposalInfo.oracles.len)
-      .map(({ transmitter }) => payeeByTransmitter[new PublicKey(transmitter).toString()])
+      .map(({ transmitter }) => input?.payeeByTransmitter[new PublicKey(transmitter).toString()])
 
-    const ix = program.instruction.proposePayees(token.publicKey, payees, {
+    const ix = this.program.instruction.proposePayees(token.publicKey, payees, {
       accounts: {
         proposal,
         authority: signer,
       },
     })
-    logger.log('Payees information:', input)
-    logger.log('Setting the following:', payees)
-
     return [ix]
   }
 
+  beforeExecute = async (input: Input) => {
+    const proposal = new PublicKey(input.proposalId)
+    const proposalInfo = await this.program.account.proposal.fetch(proposal)
+    const payeesInProposal = proposalInfo.oracles.xs.slice(0, proposalInfo.oracles.len.toNumber()).reduce(
+      (agg, { transmitter, payee }, idx) => ({
+        ...agg,
+        [`operator#${idx}`]: {
+          transmitter: transmitter.toString(),
+          payee: payee.toString(),
+        },
+      }),
+      {},
+    )
+
+    const proposedPayees = Object.entries(payeesInProposal).reduce(
+      (agg, [key, value]) => ({
+        ...agg,
+        [key]: {
+          transmitter: (value as any).transmitter,
+          payee: input?.payeeByTransmitter[(value as any).transmitter].toString(),
+        },
+      }),
+      {},
+    )
+
+    diff.printDiff(payeesInProposal, proposedPayees)
+  }
+
   execute = async () => {
-    const rawTx = await this.makeRawTransaction(this.wallet.publicKey)
+    const signer = this.wallet.publicKey
+    const input = this.makeInput(this.flags.input)
+    await this.beforeExecute(input)
+
+    const rawTx = await this.makeRawTransaction(signer, input)
+    await this.simulateTx(signer, rawTx)
     await prompt('Continue setting payees proposal?')
```

Comment:
Let's remove this and use the `beforeExecute` action only

Reply:
Please check the rest of commands in this PR, make sure we use `beforeExecute` as user confirmation